### ISSUE TITLE
AWS Toolkit causes a .vscode/settings.json file to be created or modified in every folder

### ISSUE THREAD

**Issue by kevinmarty** [aws/aws-toolkit-vscode#1933]
**Describe the bug**

Every directory opened in VS Code while the AWS Toolkit 1.28 is installed gets a .vscode/settings.json created or updated with a "yaml.customTags" section added whether or not that directory has anything to do with AWS.

**To Reproduce**

1. Create an empty directory
2. Use the Open Folder command to open it in VS Code
3. Note that almost immediately a .vscode folder with settings.json containing the following appears:

```yaml
{
    "yaml.customTags": [
        "!And",
        "!And sequence",
        "!If",
        "!If sequence",
        "!Not",
        "!Not sequence",
        "!Equals",
        "!Equals sequence",
        "!Or",
        "!Or sequence",
        "!FindInMap",
        "!FindInMap sequence",
        "!Base64",
        "!Join",
        "!Join sequence",
        "!Cidr",
        "!Ref",
        "!Sub",
        "!Sub sequence",
        "!GetAtt",
        "!GetAZs",
        "!ImportValue",
        "!ImportValue sequence",
        "!Select",
        "!Select sequence",
        "!Split",
        "!Split sequence"
    ]
}
```

**Expected behavior**

Don't create / modify the per-directory settings unless the directory is actually an AWS project.

**Desktop (please complete the following information):**

-   OS: Windows 10.0.19043
-   Visual Studio Code Version: 1.59.0
-   AWS Toolkit Version: 1.28.0

**Comment by dobeerman** [comment]
Confirmed.
It also creates the folder `file:<space>/User/....` in the root of the project opened.

**Comment by justinmk3** [comment]
Fixed in AWS Toolkit 1.29
